File: hopmux/theme.py
"""Color themes for hopmux.

Two goals from the brief: (1) simple + modern, (2) it's a *terminal* app, so it
must not be monochrome — Claude Code shows colored output and we echo that.

We register a matched dark and light theme. Accent roles are consistent across
both so the UI reads the same in either mode; only the surface/foreground flips.

Agent colors:
    claude  -> warm coral/orange  (Claude's own accent family)
    codex   -> cyan/teal
    tmux    -> green (live terminals)
    unreachable -> dim red
"""
from __future__ import annotations

from textual.theme import Theme

# Anthropic-ish coral for Claude, cyan for Codex, green for live tmux.
CLAUDE = "#d97757"
CODEX = "#2bb6c4"
TMUX = "#3fb950"
DANGER = "#f85149"

hopmux_dark = Theme(
    name="hopmux-dark",
    primary=CLAUDE,
    secondary=CODEX,
    accent=CLAUDE,
    success=TMUX,
    warning="#d29922",
    error=DANGER,
    foreground="#e6e6e6",
    background="#0d1017",
    surface="#12161f",
    panel="#171c26",
    dark=True,
    variables={
        "block-cursor-foreground": "#0d1017",
        "block-cursor-background": CLAUDE,
        "border": "#2a313c",
        "scrollbar": "#2a313c",
        "footer-key-foreground": CODEX,
        "footer-description-foreground": "#9aa4b2",
    },
)

hopmux_light = Theme(
    name="hopmux-light",
    primary="#c25b3a",
    secondary="#0e8fa0",
    accent="#c25b3a",
    success="#1a7f37",
    warning="#9a6700",
    error="#cf222e",
    foreground="#1c2128",
    background="#fbfbfa",
    surface="#f3f3f1",
    panel="#ecebe8",
    dark=False,
    variables={
        "block-cursor-foreground": "#fbfbfa",
        "block-cursor-background": "#c25b3a",
        "border": "#d7d5d0",
        "scrollbar": "#cfcdc8",
        "footer-key-foreground": "#0e8fa0",
        "footer-description-foreground": "#57606a",
    },
)


def agent_color(agent: str) -> str:
    return {"claude": CLAUDE, "codex": CODEX}.get(agent, TMUX)

File: internal/ui/theme.go
package ui

import "github.com/charmbracelet/lipgloss"

// Palette holds every color the UI uses, resolved for one mode (dark or light).
// Agent accent roles stay consistent across modes; only surfaces/foreground flip.
type Palette struct {
	Dark bool

	Fg       lipgloss.Color // primary text
	Muted    lipgloss.Color // secondary text
	Bg       lipgloss.Color // window background
	Surface  lipgloss.Color // sidebar background
	Panel    lipgloss.Color // header/footer background
	Border   lipgloss.Color

	Claude  lipgloss.Color // coral
	Codex   lipgloss.Color // cyan
	Tmux    lipgloss.Color // green
	Danger  lipgloss.Color // red
	Warning lipgloss.Color // amber (needs-auth)
	SelBg   lipgloss.Color // selection background
}

func darkPalette() Palette {
	return Palette{
		Dark:    true,
		Fg:      "#e6e6e6",
		Muted:   "#8b95a3",
		Bg:      "#0d1017",
		Surface: "#12161f",
		Panel:   "#171c26",
		Border:  "#2a313c",
		Claude:  "#d97757",
		Codex:   "#2bb6c4",
		Tmux:    "#3fb950",
		Danger:  "#f85149",
		Warning: "#d6a15e",
		SelBg:   "#243040",
	}
}

func lightPalette() Palette {
	return Palette{
		Dark:    false,
		Fg:      "#1c2128",
		Muted:   "#57606a",
		Bg:      "#fbfbfa",
		Surface: "#f2f1ee",
		Panel:   "#e9e8e4",
		Border:  "#d7d5d0",
		Claude:  "#c25b3a",
		Codex:   "#0e8fa0",
		Tmux:    "#1a7f37",
		Danger:  "#cf222e",
		Warning: "#9a6a1f",
		SelBg:   "#e2dfd8",
	}
}

// AgentColor returns the accent for a given agent name.
func (p Palette) AgentColor(agent string) lipgloss.Color {
	switch agent {
	case "claude":
		return p.Claude
	case "codex":
		return p.Codex
	default:
		return p.Tmux
	}
}
